frb_codegen/src/generator/dart/ty_struct.rs

```diff
@@ -55,24 +81,45 @@ impl TypeDartGeneratorTrait for TypeStructRefGenerator<'_> {
         let src = self.ir.get(self.context.ir_file);
         let s = self.ir.get(self.context.ir_file);
 
-        let mut methods = self.context.ir_file.funcs.iter().filter(|f| {
+        let mut methods = self.context.ir_file.funcs(true).into_iter().filter(|f| {
             let f = FunctionName::deserialize(&f.name);
             f.is_method_for_struct(&src.name) || f.is_static_method_for_struct(&src.name)
         });
         let has_methods = methods.next().is_some();
-        let mut inner = s
-            .fields
-            .iter()
-            .enumerate()
-            .map(|(idx, field)| {
-                format!(
-                    "{}: _wire2api_{}(arr[{}]),",
-                    field.name.dart_style(),
-                    field.ty.safe_ident(),
-                    idx
-                )
-            })
-            .collect::<Vec<_>>();
+
+        let mut inner = if !self.context.config.shared {
+            s.fields
+                .iter()
+                .enumerate()
+                .map(|(idx, field)| {
+                    let prefix = if !self.is_type_shared(&field.ty) {
+                        "_"
+                    } else {
+                        "_sharedImpl."
+                    };
+                    format!(
+                        "{}: {}wire2api_{}(arr[{}]),",
+                        field.name.dart_style(),
+                        prefix,
+                        field.ty.safe_ident(),
+                        idx
+                    )
+                })
+                .collect::<Vec<_>>()
+        } else {
+            s.fields
+                .iter()
+                .enumerate()
+                .map(|(idx, field)| {
+                    format!(
+                        "{}: wire2api_{}(arr[{}]),", //no prefix in shared block
+                        field.name.dart_style(),
+                        field.ty.safe_ident(),
+                        idx
+                    )
+                })
+                .collect::<Vec<_>>()
+        };
```

Comment:
nit (code is not formatted since I type in github's webpage)

```suggestion
        let mut inner = 
            s.fields
                .iter()
                .enumerate()
                .map(|(idx, field)| {
                    let prefix = if !self.context.config.shared {
                    if !self.is_type_shared(&field.ty) {
                        "_"
                    } else {
                        "_sharedImpl."
                    }
                    } else { "" };
                    format!(
                        "{}: {}wire2api_{}(arr[{}]),",
                        field.name.dart_style(),
                        prefix,
                        field.ty.safe_ident(),
                        idx
                    )
                })
                .collect::<Vec<_>>()
```

Reply:
solved in 9dba548.